### PR TITLE
Add ServiceInstanceTopology & ServiceInstanceNode

### PR DESCRIPTION
**Pull request by arugal**
Ref: [apache/skywalking#3950](https://github.com/apache/skywalking/pull/3978)

### CODE REVIEW

File: topology.graphqls [hunk 1]
```diff
@@ -36,6 +42,26 @@ type Node {
...
+    # 1. Instance id
+    instanceId: ID!
+    # The literal name of the #instanceId.
+    instanceName: String!
+    # The type name may be
+    # 1. The service provider/middleware tech, such as: Tomcat, SpringMVC
+    # 2. Conjectural Service, e.g. MySQL, Redis, Kafka
+    type: String
+    # It is a conjuecture node or real node, to represent a service or endpoint.
+    isReal: Boolean!
```
Comment (wu-sheng) [inline topology.graphqls]: Instance node represent an instance, not service or endpoint.

File: topology.graphqls [hunk 2]
```diff
@@ -36,6 +42,26 @@ type Node {
     isReal: Boolean!
 }
 
+# Node in ServiceInstanceTopology
+type ServiceInstanceNode {
+    # The global id of each node,
```
Comment (wu-sheng) [inline topology.graphqls]: This is not the ID.

File: topology.graphqls [hunk 3]
```diff
@@ -36,6 +42,26 @@ type Node {
...
+    # 1. Service id
+    serviceId: ID!
+    # The literal name of the #serviceId.
+    serviceName: String!
+    # The global id of each node,
+    # 1. Instance id
+    instanceId: ID!
```
Comment (wu-sheng) [inline topology.graphqls]: This should be renamed as id, like `Node#id`. Becuase this is used on `Call` object.

File: topology.graphqls [hunk 4]
```diff
@@ -36,6 +42,26 @@ type Node {
...
+    # The literal name of the #instanceId.
+    instanceName: String!
```
Comment (wu-sheng) [inline topology.graphqls]: Should be renamed as `name`.

**Comment by arugal** [comment]
@wu-sheng Completed, please review.

**Comment by wu-sheng** [comment]
Done. You could work on main repo PR now.

**Pull request merged**